src/private/lidar_impl.cpp

```diff
@@ -0,0 +1,191 @@
+#include "context_impl.cpp"
+#include <vector>
+#include <libm2k/utils/utils.hpp>
+#include <libm2k/utils/channel.hpp>
+#include <libm2k/utils/devicegeneric.hpp>
+#include <libm2k/utils/devicein.hpp>
+#include <libm2k/lidar.hpp>
+#include <iio.h>
+#include <iostream>
+
+using namespace std;
+using namespace libm2k::contexts;
+using namespace libm2k::utils;
+
+class Lidar::LidarImpl : public ContextImpl {
+public:
+	LidarImpl(std::string uri, iio_context* ctx, std::string name, bool sync) :
+		ContextImpl(uri, ctx, name, sync) {
+
+		pulse_dev = new DeviceGeneric(ctx, "7c700000.axi-pulse-capture");
+		afe_dev	  = new DeviceGeneric(ctx, "ad5627");
+		adc_dev	  = new DeviceIn     (ctx, "axi-ad9094-hpc");
+
+		setAllIIoAttributesToDefaultValues();
+	}
+
+	void initialize() {
+		;
+	}
+    
+	void channelEnableDisable(string channel, bool enable) {
+		if	(channel == "voltage0")
+			adc_dev->enableChannel(0, enable, false);
+		else if (channel == "voltage1")
+			adc_dev->enableChannel(1, enable, false);
+		else if (channel == "voltage2")
+			adc_dev->enableChannel(2, enable, false);
+		else if (channel == "voltage3")
+			adc_dev->enableChannel(3, enable, false);
+		else if (channel == "voltage4")
+			adc_dev->enableChannel(4, enable, false);
+		else
+			return;		    // invalid channel
+	}
+
+	vector<string> enabledChannelsList() {
+		vector<string> enabled_channels;
+		if (adc_dev->isChannelEnabled(0, false))
```

Comment:
maybe do for loop here as well?
```
```
for (i = 0; i < MAX_CHANELS; i++) {
     if (if (adc_dev->isChannelEnabled(i, false)) {
       enabled_channels.push_back("voltage" + i); 
    }
```

Reply:
Same as above.